data/src/main/java/com/microsoft/azure/kusto/data/ClientRequestProperties.java

```diff
@@ -70,25 +77,51 @@ public Long getTimeoutInMilliSec() {
         if (timeoutObj instanceof Long) {
             timeout = (Long) timeoutObj;
         } else if (timeoutObj instanceof String) {
-            timeout = LocalTime.parse((String) timeoutObj).toNanoOfDay() / NANOS_TO_MILLIS;
+            timeout = getMillisFromTimespanString((String) timeoutObj);
         } else if (timeoutObj instanceof Integer) {
             timeout = Long.valueOf((Integer) timeoutObj);
         }
 
         return timeout;
     }
 
+    private long getMillisFromTimespanString(String str) throws TimespanParseException {
+        Matcher matcher = PATTERN.matcher(str);
+        if (!matcher.matches()) {
+            throw new TimespanParseException(str);
+        }
+
+        long millis = 0;
+        String days = matcher.group(1);
+        if (days != null) {
+            millis += Long.parseLong(days) * SECONDS_PER_DAY * 1000L;
+        }
+
+        millis += LocalTime.parse(matcher.group(2)).toNanoOfDay() / 1000000L;
+        return millis;
+    }
+
     public void setTimeoutInMilliSec(Long timeoutInMs) {
         options.put(OPTION_SERVER_TIMEOUT, timeoutInMs);
     }
 
     JSONObject toJson() {
         try {
             JSONObject optionsAsJSON = new JSONObject(this.options);
-            Long timeoutInMilliSec = getTimeoutInMilliSec();
-            if (timeoutInMilliSec != null) {
-                LocalTime localTime = LocalTime.ofNanoOfDay(timeoutInMilliSec * NANOS_TO_MILLIS);
-                optionsAsJSON.put(OPTION_SERVER_TIMEOUT, localTime.toString());
+            Object timeoutObj = getOption(OPTION_SERVER_TIMEOUT);
+
+            if (timeoutObj != null) {
+                String timeoutString = "";
+                if (timeoutObj instanceof Long) {
```

Comment:
I don't think this would work. Wouldn't every value in a JSON be interpreted as a String?

I think all 3 of:
- {"Options":{"servertimeout":"2"}}
- {"Options":{"servertimeout":"9999999999"}}
- {"Options":{"servertimeout":"445635.29:60:60.354654"}}
Would resolve as `instanceof String`

Reply:
I wrote a test to check those other values, and as far as I could tell, in all cases it interpreted the object as a String.